<?xml version="1.0"?>
<launch>
  <arg name="model" default="$(find example-robot-data)/robots/iris_description/robots/iris_simple.urdf"/>
  <param name="robot_description" textfile="$(arg model)" />
  
  <node pkg="multicopter_mpc_viz" name="trajectory_generator" type="trajectory_generation_rviz.py" output="screen">
    <remap from="/whole_body_state" to="/trajectory_generator/whole_body_state"/>
    <remap from="/whole_body_trajectory" to="/trajectory_generator/whole_body_trajectory"/>
  </node>

  <node pkg="multicopter_mpc_viz" name="mpc_controller" type="mpc_controller_rviz.py" output="screen">
    <remap from="/whole_body_state" to="/mpc_controller/whole_body_state"/>
    <remap from="/whole_body_trajectory" to="/mpc_controller/whole_body_trajectory"/>
    <remap from="/whole_body_partial_trajectory" to="/mpc_controller/whole_body_partial_trajectory"/>
    <remap from="/parameter_updates" to="/trajectory_generator/parameter_updates"/>    
  </node>

  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find multicopter_mpc_ros)/rviz/mpc_controller_display.rviz" />

  <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="trajectory_rqt_reconfigure"/>


</launch>